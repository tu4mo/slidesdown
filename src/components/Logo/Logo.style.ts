import styled from 'styled-components'

interface Props {
  large?: boolean
}

export const StyledImg = styled.img<Props>`
  cursor: ${(props) => props.onClick && 'pointer'};
  display: block;
  height: ${(props) => (props.large ? '5rem' : '2rem')};
  width: ${(props) => (props.large ? '5rem' : '2rem')};
`
