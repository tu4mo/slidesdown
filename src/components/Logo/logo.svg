<svg width="40" height="40" xmlns="http://www.w3.org/2000/svg"><defs><linearGradient x1="0%" y1="0%" x2="102%" y2="101%" id="a"><stop stop-color="#3023AE" offset="0%"/><stop stop-color="#C86DD7" offset="100%"/></linearGradient></defs><g fill-rule="nonzero" fill="none"><circle fill="url(#a)" cx="20" cy="20" r="20"/><path fill="#FFF" d="M25 12.5h4v10h-4z"/><path fill="#FFF" d="M27 28l-6-6h12zM14.086 15.032c-.576 0-1.03.136-1.361.41-.332.272-.498.634-.498 1.084 0 .45.203.807.608 1.073.406.265 1.343.579 2.81.94 1.469.362 2.609.904 3.42 1.627.811.723 1.217 1.778 1.217 3.165 0 1.386-.52 2.511-1.56 3.374S16.314 28 14.617 28c-2.45 0-4.655-.907-6.617-2.722l2.058-2.523c1.667 1.46 3.209 2.191 4.625 2.191.635 0 1.133-.136 1.494-.41.361-.272.542-.641.542-1.106a1.29 1.29 0 0 0-.575-1.106c-.384-.273-1.144-.55-2.28-.83-1.8-.428-3.116-.985-3.95-1.671-.833-.686-1.25-1.763-1.25-3.231s.527-2.6 1.582-3.397c1.055-.797 2.372-1.195 3.95-1.195 1.033 0 2.066.177 3.099.531 1.032.354 1.932.856 2.7 1.505l-1.749 2.523c-1.342-1.018-2.73-1.527-4.16-1.527z"/></g></svg>
