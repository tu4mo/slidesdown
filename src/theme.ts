export default {
  breakpoints: {
    md: '576px',
    xl: '768px'
  },
  colors: {
    background: '#edf1f5',
    border: '#e4e4e8',
    dimText: '#a4a4a8',
    lightText: '#646468',
    magenta: '#c86dd7',
    purple: '#3023ae'
  },
  fonts: {
    default: 'Rubik',
    monospace: 'Inconsolata'
  }
}
