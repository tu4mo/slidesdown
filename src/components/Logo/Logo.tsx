import { forwardRef } from 'react'

import { StyledImg } from './Logo.style'

import logoSvg from './logo.svg'

interface Props {
  large?: boolean
  onClick?(): void
}

const Logo = forwardRef<HTMLImageElement, Props>(({ large, onClick }, ref) => (
  <StyledImg
    $large={large}
    alt="Slidesdown"
    onClick={onClick}
    ref={ref}
    src={logoSvg}
  />
))

Logo.displayName = 'Logo'

export { Logo }
