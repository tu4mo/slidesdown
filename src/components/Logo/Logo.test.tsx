import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'

import theme from '../../theme'
import Logo from './Logo'

jest.mock('./logo.svg', () => 'Logo')

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Logo />
      </ThemeProvider>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
