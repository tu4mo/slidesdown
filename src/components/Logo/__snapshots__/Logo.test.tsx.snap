// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: block;
  height: 2rem;
  width: 2rem;
}

<img
  alt="Slidesdown"
  className="c0"
  src="Logo"
/>
`;
